pkgs/by-name/ro/rollo-x1038/package.nix

```
@@ -0,0 +1,30 @@
+{
+  stdenv,
+  cups,
+  lib,
+}:
+
+stdenv.mkDerivation {
+  name = "rollo-x1038";
+  version = "1.8.4";
+  src = fetchTarball {
+    url = "https://rollo-main.b-cdn.net/driver-dl/linux/rollo-cups-driver-1.8.4.tar.gz";
+    sha256 = "sha256:0x8dd6wj5n974k3vb3iir61dd46ag13zj7s9dhgv722cky7v57h4";
+  };
+  nativeBuildInputs = [
+    cups
+  ];
+  installPhase = ''
+    install -d -m 755 $out/lib/cups/filter
+    install -c -m 755 rastertorollo $out/lib/cups/filter
+    install -d -m 755 $out/share/cups/model
+    install -c -m 644 rollo-x1038.ppd $out/share/cups/model
+  '';
```

```suggestion
  installPhase = ''
    runHook preInstall

    install -Dm755 rastertorollo -t $out/lib/cups/filter
    install -Dm644 rollo-x1038.ppd -t $out/share/cups/model

    runHook postInstall
  '';
```
1. Phases must remain adjustable (hence `runHook`)
2. These commands can be trimmed to just two